### ISSUE TITLE
Phase 1: Basic MLflow Integration and Setup

### ISSUE THREAD

**Issue by adhit-r** [adhit-r/RagaSense#31]
### Overview
Set up the foundation for MLflow integration by installing, configuring, and converting our existing ML pipelines to use MLflow tracking.

### Current ML Pipeline Status
- Enhanced training pipeline: 106 features, 100% accuracy on 37 samples
- Tradition classification: 21 cultural features, 100% accuracy on 200 samples
- Manual model saving and versioning
- No experiment tracking or reproducibility

#### 1. MLflow Installation and Configuration
- [ ] Install MLflow and dependencies
- [ ] Configure local tracking URI (file:./mlruns)
- [ ] Set up basic MLflow environment
- [ ] Test MLflow installation and basic functionality

#### 2. Enhanced Training Pipeline Conversion
- [ ] Convert scripts/enhanced_training_pipeline.py to use MLflow
- [ ] Add experiment tracking for training runs
- [ ] Log training parameters (feature_count, sample_rate, duration, algorithm)
- [ ] Log training metrics (accuracy, cv_score, training_time)
- [ ] Save models to MLflow model registry
- [ ] Test MLflow-enhanced training pipeline

#### 3. Tradition Classification Pipeline Conversion
- [ ] Convert scripts/tradition_classification_system.py to use MLflow
- [ ] Add experiment tracking for tradition classification
- [ ] Log cultural feature parameters (gamaka, meend, shruti, rhythm, melodic, timbre, performance)
- [ ] Log classification metrics (accuracy, cv_score, tradition_distribution)
- [ ] Save tradition classifier to MLflow model registry
- [ ] Test MLflow-enhanced tradition classification

#### 4. Basic Experiment Tracking
- [ ] Create MLflow experiments for different feature extraction approaches
- [ ] Track model performance across different algorithms
- [ ] Implement basic experiment comparison
- [ ] Test experiment reproducibility

#### File Structure Changes
```
ml/
├── mlflow_projects/           # NEW: MLflow project structure
│   ├── enhanced_training/
│   │   ├── MLproject
│   │   ├── conda.yaml
│   │   └── train.py          # Converted from scripts/
│   └── tradition_classification/
│       ├── MLproject
│       ├── conda.yaml
│       └── train.py          # Converted from scripts/
├── mlruns/                   # NEW: MLflow tracking data
└── models/                   # NEW: Registered models
```

#### Code Changes Required

| File | Current | Target | Changes Needed |
|------|---------|--------|----------------|
| enhanced_training_pipeline.py | Manual training | MLflow tracking | Add mlflow.start_run(), log_param(), log_metric(), log_model() |
| tradition_classification_system.py | Manual training | MLflow tracking | Add mlflow.start_run(), log_param(), log_metric(), log_model() |
| requirements.txt | Basic ML | MLflow + ML | Add mlflow, mlflow[extras] |

#### MLflow Integration Points

| Component | Integration Method | Priority |
|-----------|-------------------|----------|
| Experiment Tracking | mlflow.set_experiment() | High |
| Parameter Logging | mlflow.log_param() | High |
| Metric Logging | mlflow.log_metric() | High |
| Model Saving | mlflow.sklearn.log_model() | High |
| Artifact Logging | mlflow.log_artifact() | Medium |

#### New Dependencies
```bash
pip install mlflow
pip install mlflow[extras]
```

#### System Requirements
- Python 3.9+
- 2GB+ free disk space for MLflow tracking
- Existing ML pipeline scripts functional

#### Functional Requirements
- [ ] MLflow installation successful
- [ ] Enhanced training pipeline runs with MLflow tracking
- [ ] Tradition classification pipeline runs with MLflow tracking
- [ ] Models saved to MLflow model registry
- [ ] Experiments trackable via MLflow UI

#### Quality Requirements
- [ ] All existing functionality preserved
- [ ] MLflow tracking adds <5% overhead to training time
- [ ] Experiment data properly organized and searchable
- [ ] Models loadable from MLflow registry

#### Unit Tests
- [ ] Test MLflow installation and configuration
- [ ] Test enhanced training pipeline with MLflow
- [ ] Test tradition classification with MLflow
- [ ] Test model loading from registry

#### Integration Tests
- [ ] Test end-to-end MLflow pipeline
- [ ] Test experiment tracking and comparison
- [ ] Test model versioning and deployment

#### Code Deliverables
- [ ] MLflow-enhanced training pipeline
- [ ] MLflow-enhanced tradition classification
- [ ] MLflow project configuration files
- [ ] Updated requirements and dependencies

#### Documentation Deliverables
- [ ] MLflow setup and configuration guide
- [ ] Updated ML pipeline documentation
- [ ] MLflow usage examples and best practices
- [ ] Migration guide from manual to MLflow pipelines

### Risk Assessment

| Risk | Probability | Impact | Mitigation |
|------|-------------|--------|------------|
| MLflow installation issues | Low | Medium | Test on clean environment, document dependencies |
| Performance degradation | Medium | High | Benchmark before/after, optimize MLflow calls |
| Model compatibility issues | Low | High | Test model loading, maintain backward compatibility |
| Learning curve for team | Medium | Medium | Provide training, start with simple examples |

#### Week 1 Breakdown
- **Days 1-2**: MLflow installation and configuration
- **Days 3-4**: Enhanced training pipeline conversion
- **Days 5-7**: Tradition classification pipeline conversion
- **Day 7**: Testing and validation

### Next Phase Dependencies
- Phase 1 completion required for Phase 2 (Advanced Pipeline Management)
- Basic MLflow setup required for Phase 3 (Production Integration)
- Experiment tracking foundation required for advanced features

### Notes
- Focus on functionality over optimization in Phase 1
- Maintain backward compatibility with existing scripts
- Document all MLflow integration points for team reference
- Consider creating MLflow templates for future pipeline development

**Comment by adhit-r** [comment]
Closing duplicate - consolidating roadmap
